migrace/2024-06-28_uzitecne-funkce.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/** @var \Godric\DbMigrations\Migration $this */
+
+$this->q(<<<SQL
+CREATE FUNCTION IF NOT EXISTS aktualniRocnik() RETURNS int READS SQL DATA
+RETURN (select hodnota from gamecon.systemove_nastaveni where klic = 'ROCNIK' limit 1)
+SQL
+);
+
+$this->q(<<<SQL
+CREATE FUNCTION IF NOT EXISTS systemoveNastaveni(klic varchar(128)) RETURNS varchar(255) READS SQL DATA
+    RETURN (select hodnota from gamecon.systemove_nastaveni sn where sn.klic collate utf8mb3_general_ci = klic limit 1)
```

Comment:
Ten collate tu potřebujeme? Všechny klíče jsou ASCII only, takže pro UTF8 by mělo být collation úplně jedno. Proč vlastně tady potřebujeme explicitní collation a nestačí to které nastavujeme při připojování k databázi ? :thinking:

Reply:
Tam byl nějaký bullshit nebo něco AFAIK s tím že ten sloupeček je kódovaný v latince a funkce je kódovaná by default UTF a statická analýza neví že je to úplně jedno, protože data budou vžycky ASCII, protože to nemá jak vědět, takže tam někde tu explicitní collaci vyžaduje